File: 백준/Silver/8911. 거북이/거북이.py
T = int(input())

for _ in range(T):
    rect = {
        "left_x": 0,
        "right_x": 0,
        "top_y": 0,
        "bottom_y": 0,
    }
    locate = [0, 0]
    # y축 상숭 -> x축 상승 -> y축 하강 -> x축 하강
    dir = [[0, 1], [1, 0], [0, -1], [-1, 0]]
    d = 0
    orders = list(input())
    for o in orders:
        # F: 한 눈금 앞으로
        if o == "F":
            locate[0] += dir[d][0]
            locate[1] += dir[d][1]
        # B: 한 눈금 뒤로
        elif o == "B":
            locate[0] -= dir[d][0]
            locate[1] -= dir[d][1]
        # L: 왼쪽으로 90도 회전
        elif o == "L":
            if d == 0:
                d = 3
            else:
                d -= 1
        # R: 오른쪽으로 90도 회전
        elif o == "R":
            if d == 3:
                d = 0
            else:
                d += 1
        # 갱신

        rect["left_x"] = min(rect["left_x"],locate[0])
        rect["right_x"] = max(rect["right_x"], locate[0])
        rect["bottom_y"] = min(rect["bottom_y"], locate[1])
        rect["top_y"] = max(rect["top_y"], locate[1])

    print((rect["right_x"]-rect["left_x"]) * (rect["top_y"]-rect["bottom_y"]))

File: 백준/Silver/8911. 거북이/거북이.java
import java.util.*;

public class Main {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        // 입력
        int T = sc.nextInt();
        sc.nextLine(); // 버퍼를 비웁니다.

        for (int t = 0; t < T; t++) {
            int leftX = 0;
            int rightX = 0;
            int topY = 0;
            int bottomY = 0;
            int[] nowLocation = {0, 0};
            // y축 상승 -> x축 상승 -> y축 하강 -> x축 하강
            int[][] dir = {{0, 1}, {1, 0}, {0, -1}, {-1, 0}};
            int d = 0;

            String orders = sc.nextLine();

            for (int i = 0; i < orders.length(); i++) {
                char o = orders.charAt(i);
                // F: 한 눈금 앞으로
                if (o == 'F') {
                    nowLocation[0] += dir[d][0];
                    nowLocation[1] += dir[d][1];
                }
                // B: 한 눈금 뒤로
                else if (o == 'B') {
                    nowLocation[0] -= dir[d][0];
                    nowLocation[1] -= dir[d][1];
                }
                // L: 왼쪽으로 90도 회전
                else if (o == 'L') {
                    if (d == 0) {
                        d = 3;
                    } else {
                        d -= 1;
                    }
                }
                // R: 오른쪽으로 90도 회전
                else if (o == 'R') {
                    if (d == 3) {
                        d = 0;
                    } else {
                        d += 1;
                    }
                }

                leftX = Math.min(nowLocation[0], leftX);
                rightX = Math.max(nowLocation[0], rightX);
                topY = Math.max(nowLocation[1], topY);
                bottomY = Math.min(nowLocation[1], bottomY);
            }

            int area = (rightX - leftX) * (topY - bottomY);

//            System.out.printf("(%d - %d) - (%d - %d) = %d \n", rightX, leftX, topY, bottomY, area);
            System.out.println(area);
        }
    }
}
